digdag-standards/src/main/java/io/digdag/standards/operator/HttpOperatorFactory.java

```diff
@@ -0,0 +1,389 @@
+                    request.getConfig().getNestedOrGetEmpty("http"));
+        }
+
+        @Override
+        public List<String> secretSelectors()
+        {
+            return ImmutableList.of("http.*");
+        }
+
+        @Override
+        public TaskResult runTask(TaskExecutionContext ctx)
+        {
+            HttpClient client = client();
+            try {
+                return run(ctx, client);
+            }
+            finally {
+                stop(client);
+            }
+        }
+
+        private TaskResult run(TaskExecutionContext ctx, HttpClient httpClient)
+        {
+            // TODO: support secrets in headers, uri fragments, payload fragments, etc
+
+            SecretProvider httpSecrets = ctx.secrets().getSecrets("http");
+
+            URI uri = URI.create(httpSecrets.getSecretOptional("uri").or(() -> params.get("_command", String.class)));
+            String method = params.get("method", String.class, "GET");
+
+            Optional<String> user = httpSecrets.getSecretOptional("user");
+            Optional<String> authorization = httpSecrets.getSecretOptional("authorization");
+
+            Request request = httpClient.newRequest(uri)
+                    .method(method)
+                    .timeout(30, SECONDS);
+
+            if (authorization.isPresent()) {
+                request.header(AUTHORIZATION, authorization.get());
+            }
+            else if (user.isPresent()) {
+                Optional<String> password = httpSecrets.getSecretOptional("password");
+                httpClient.getAuthenticationStore().addAuthenticationResult(new BasicAuthentication.BasicResult(uri, user.get(), password.or("")));
+            }
+
+            Optional<String> content = params.getOptional("content", String.class);
+            Optional<String> contentType = params.getOptional("content_type", String.class);
+            boolean storeContent = params.get("store_content", boolean.class, false);
+
+            if (content.isPresent()) {
+                // TODO: support POST url encoded key/val and JSON encoding nested objects, files on disk etc
+                ContentProvider contentProvider = new StringContentProvider(content.get());
+                request.content(contentProvider, contentType.orNull());
+            }
+
+            LinkedHashMultimap<String, String> headers = headers();
+            for (Map.Entry<String, String> header : headers.entries()) {
+                request.header(header.getKey(), header.getValue());
+            }
+
+            ContentResponse response = PollingRetryExecutor.pollingRetryExecutor(state, "request")
+                    .withErrorMessage("HTTP request failed")
+                    .run(s -> execute(request));
+
+            return result(response, storeContent);
+        }
+
+        private LinkedHashMultimap<String, String> headers()
+        {
+            List<JsonNode> entries = params.getListOrEmpty("headers", JsonNode.class);
```

Comment:
These are valid headers:

```
headers:
  - Foo: foo-1
  - Bar: bar
  - Foo: foo-2
```

And the http client should preserve header ordering. That is not possible with a simple object.